pkg/xds/sync/dataplane_watchdog.go

```diff
@@ -96,7 +97,19 @@ func (d *DataplaneWatchdog) Cleanup() error {
 	case mesh_proto.IngressProxyType:
 		return d.IngressReconciler.Clear(&proxyID)
 	case mesh_proto.EgressProxyType:
-		return d.EgressReconciler.Clear(&proxyID)
+		var errs []error
+		aggregatedMeshCtxs, err := xds_context.AggregateMeshContexts(
+			context.TODO(),
+			d.ResManager,
+			d.MeshCache.GetMeshContext,
+		)
+		errs = append(errs, err)
+		for _, mesh := range aggregatedMeshCtxs.Meshes {
+			d.EnvoyCpCtx.Secrets.Cleanup(core_model.ResourceKey{Mesh: mesh.GetMeta().GetName(), Name: d.key.Name})
+		}
+		err = d.EgressReconciler.Clear(&proxyID)
+		errs = append(errs, err)
+		return std_errors.Join(errs...)
```

Comment:
Just a nit, there are only two errors, so maybe be could just do
```suggestion
		aggregatedMeshCtxs, aggregateMeshContextsErr := xds_context.AggregateMeshContexts(
			context.TODO(),
			d.ResManager,
			d.MeshCache.GetMeshContext,
		)
		for _, mesh := range aggregatedMeshCtxs.Meshes {
			d.EnvoyCpCtx.Secrets.Cleanup(core_model.ResourceKey{Mesh: mesh.GetMeta().GetName(), Name: d.key.Name})
		}
		return std_errors.Join(aggregateMeshContextsErr, d.EgressReconciler.Clear(&proxyID))
```